### DISCUSSION TITLE
i have a keymap-editor Firmware problem

### DISCUSSION THREAD

**Discussion by koreatwosome** [nickcoutsos/keymap-editor#178] [Q&A] [answered]
First of all, thank you for the useful open sourve you created.
I'm using a customized keymap using the keymap-editor you made on the Corne keyboard. 

But at some point, no new keymap firmware is being created. Only existing firmware is visible in the 'Actions catagory'(It looks like the firmware is actually being created for new keymapping, but it's the existing firmware. Maybe it's a compilation error)

I wonder if there is a presumptive cause for this problem

sincerely
from. HOYA

![Screenshot_20240123_172925_Gallery](https://github.com/nickcoutsos/keymap-editor/assets/157190071/b4941da0-1e1c-4260-a40f-52437bb00268)

**Comment by nickcoutsos** [comment, marked as answer]
It looks it's building successfully every time, so that's good.

Looking at your build history it looks like the build you're looking at is relatively old -- there are [8 newer builds](https://github.com/koreatwosome/cornekeymap/actions/workflows/build.yml), and GitHub Actions will only keep your resulting firmware on a limited number of the most recent builds.

> It looks like the firmware is actually being created for new keymapping, but it's the existing firmware.

By this do you mean you flashed the new firmware but they keymap is unchanged? You may need to go to the ZMK community for more support, if the keymap source compiles successfully that's kind of the limit of what this app can do.

Just to rule some things out (and I apologize if these seem obvious)

- are you sure you aren't accidentally flashing old firmware (e.g., did your browser download `firmware (1).zip` but you looked at `firmware.zip` instead)
- are you flashing only the right side of the keyboard? The keymap is only processed on the left and you can technically get away with only flashing updated firmware there in _most_ cases but it doesn't hurt to flash both for consistency.

**Reply by koreatwosome** [reply]
Thank you very much for your help.
I was able to solve the related problem easily thanks to the fact that the keymap only works on the left keyboard. :)

All conditions are fine, and the degree of freedom of key mapping has been improved.

Once again, thank you for your great help :)

sincerely
from. HOYA

**Reply by nickcoutsos** [reply]
Excellent news! Glad it worked out!
